ledger-service/http-json/src/main/scala/com/digitalasset/http/domain.scala

```diff
@@ -144,6 +157,14 @@ object domain {
 
     def toLedgerApi(o: Offset): lav1.ledger_offset.LedgerOffset =
       lav1.ledger_offset.LedgerOffset(lav1.ledger_offset.LedgerOffset.Value.Absolute(unwrap(o)))
+
+    implicit val ordering: Ordering[Offset] =
+      Ordering.by(LedgerOffsetUtil.parseOffsetString _ compose unwrap)(
+        LedgerOffsetUtil.LongEitherLongLongOrdering)
+
+    implicit val semigroup: Semigroup[Offset] = new Semigroup[Offset] {
```

Comment:
"Later in the stream" is always the desired one; comparing them is an unneeded complication.

```suggestion
    implicit val semigroup: Semigroup[Offset] = Tag.unsubst(Semigroup[Offset @@ Tags.LastVal])
```

Reply:
`Tags.LastVal` is not the same as `Ordering.max`... why LastVal? We care about the largest offset, not the last one?

Reply:
No, in this case, we care about the last one.

We care about the largest one in the only temporally-unordered scenario: that is, the one that you originally implemented the comparison for. Stream steps are already temporally ordered, so introducing another layer of ordering on top of that is needless complication.

Reply:
plus it will break the associativity law.

Reply:
No, both max and lastval are associative. That is why scalaz defines semigroups for them at all.

Reply:
a1 + (a2 + a3) = a3
(a1 + a2) + a3 = a3
scratch the associaitivity law :)

Reply:
@S11001001 thanks... you are 100% right.... if I leave the semigroup implementation as ordering.max, the property that we observe in the test will always be true:
```
lastSeen should be > liveStart
```
so, yeah we should capture the last one, the left operand.